plugin/hclog_adapter.go

```diff
@@ -29,6 +29,10 @@ func (h *hclogAdapter) Trace(msg string, args ...interface{}) {
 
 func (h *hclogAdapter) Debug(msg string, args ...interface{}) {
 	extras := strings.TrimSpace(fmt.Sprint(args...))
+	if strings.Contains(msg, "panic: ") || strings.Contains(extras, "panic: ") {
```

Comment:
This seems like logic that does not belong here. This is just an adapter from hclog to mlog. 
It's also not ideal for performance to do do string compares for every debug log.

Just an unreseached idea for an alternate: Could we possibly capture panics in `plugin/client.go` and route them appropriately?

Reply:
@iomodo and I synced up offline, and will investigate finding another location for this. It might be possible to extract this solely from the STDERR capture process, from which presumably we get a lot less output.

@iomodo, should we also constrain the string comparison? I'm guessing we can look for `panic: ` at just the start of the string, vs. having to scan the whole thing?